src/Interpreters/Cache/WriteBufferToFileSegment.cpp

```diff
@@ -134,7 +134,7 @@ std::unique_ptr<ReadBuffer> WriteBufferToFileSegment::getReadBufferImpl()
     if (file_segment->getDownloadedSize() > 0)
         return std::make_unique<ReadBufferFromFile>(file_segment->getPath());
     else
-        return std::make_unique<EmptyReadBuffer>();
+        return std::make_unique<ReadBufferFromEmptyFile>();
```

Comment:
But `prepareVerticalMergeForAllColumns` is waiting for a `ReadBufferFromFile`, not a `ReadBufferFromEmptyFile`. It's going to still fail, right?